src/Cake.Common/Tools/OctopusDeploy/PushPackageArgumentBuilder.cs

```diff
@@ -0,0 +1,42 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+using System.Collections.Generic;
+using System.Linq;
+using Cake.Core;
+using Cake.Core.IO;
+
+namespace Cake.Common.Tools.OctopusDeploy
+{
+    internal class PushPackageArgumentBuilder : OctopusDeployArgumentBuilder<PushPackageSettings>
```

Comment:
@agc93 and also, the same comment as before, deriving from the deploy class feels "wrong" to me.  Perhaps a common class, which both derive from?  I think there are examples of using this approach in the code base.  @devlead @patriksvensson thoughts?